include/swift/AST/DiagnosticsSema.def

```diff
@@ -397,6 +397,41 @@ ERROR(cannot_convert_argument_value_generic,none,
       "cannot convert value of type %0 (%1) to expected argument type %2 (%3)",
       (Type, StringRef, Type, StringRef))
 
+// @_nonEphemeral conversion diagnostics
+ERROR(cannot_pass_type_to_non_ephemeral,none,
+      "cannot pass %0 to parameter; argument #%1 must be a pointer that "
+      "outlives the call%select{| to %3}2", (Type, unsigned, bool, DeclName))
```

Comment:
This is okay but it'd be nice to get the names in here too (see `argument_out_of_order_named_named`). That can be follow-up work for sure, though.

Reply:
@jrose-apple Good idea! I'll do this as a follow-up, as we'll probably want to add a new "get argument label" API to `ArgumentMismatchFailure` and `FunctionArgApplyInfo`.